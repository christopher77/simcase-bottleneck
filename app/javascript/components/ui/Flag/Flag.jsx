import React from "react";
import styles from "./Flag.module.css";
import greenFlag from "../../../assets/green-flag.svg";
import redFlag from "../../../assets/red-flag.svg";

function Flag({ isCorrect, flagDirection, numStation }) {
  const flagImage = isCorrect ? greenFlag : redFlag;
  const classStyle =
    (flagDirection == "left" ? "flagLeft" : "flagRight") + numStation;
  return (
    <>
      <img src={flagImage} className={styles[classStyle]} />
    </>
  );
}

export default Flag;
