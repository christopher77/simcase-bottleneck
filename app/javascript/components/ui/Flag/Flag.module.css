.button {
  width: 45px;
  height: 45px;
  border-radius: 2px;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 24px;
  font-weight: 400;
  background: linear-gradient(
    180deg,
    #ffffff 0%,
    #ffffff 0.01%,
    #ffffff 47.4%,
    #ececec 100%
  );
  box-shadow: 0px 0px 4px 2px rgba(115, 192, 86, 0.8);
  border-radius: 5px;
}

.buttonCorrect {
  composes: button;
  animation: flip-vertical-fwd 1s cubic-bezier(0.455, 0.03, 0.515, 0.955) both;
}

.buttonFail {
  composes: button;
  box-shadow: 0px 0px 4px 2px rgba(201, 20, 20, 0.8);
  border-radius: 5px;
  animation: flip-vertical-fwd 1s cubic-bezier(0.455, 0.03, 0.515, 0.955) both;
}
.flagLeft1 {
  position: absolute;
  left: 18px;
  top: -48px;
}
.flagLeft2 {
  position: absolute;
  left: 4px;
  top: -49px;
}

.flagRight3 {
  position: absolute;
  transform: matrix(-0.37, -0.93, -0.93, 0.37, 0, 0) rotate(71.52deg);
  left: -4px;
  top: -48px;
}
.flagRight4 {
  position: absolute;
  transform: matrix(-0.13, -0.99, -0.99, 0.13, 0, 0) rotate(66.72deg);
  left: 9px;
  top: -49px;
}
.flagRight5 {
  position: absolute;
  transform: matrix(-0.13, -0.99, -0.99, 0.13, 0, 0) rotate(70deg);
  left: 9px;
  top: -49px;
}
