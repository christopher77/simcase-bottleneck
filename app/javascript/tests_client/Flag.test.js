import React from "react";
import { render } from "@testing-library/react";
import Flag from "../components/ui/Flag/Flag";

test("Flag component", () => {
  const { asFragment } = render(<Flag />);
  expect(asFragment()).toMatchSnapshot();
});
