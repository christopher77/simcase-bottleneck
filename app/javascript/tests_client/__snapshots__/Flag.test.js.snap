// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flag component 1`] = `
<DocumentFragment>
  <img
    class="flagRightundefined"
    src="[object Object]"
  />
</DocumentFragment>
`;
